server/client.ts

```diff
@@ -129,10 +137,15 @@ class Client {
 			highlightRegex: null,
 			highlightExceptionRegex: null,
 			messageProvider: undefined,
+			events: new ClientEmitter(),
 		});
 
 		const client = this;
 
+		for (const event in packageEvents) {
+			client.events.on(event as keyof ClientEvents, packageEvents[event]);
```

Comment:
Can you move the `as` cast up to the packageEvents declaration?

Reply:
Changed this to be a typed map to take care of this